### PR TITLE
HTCONDOR-1491 kill-sprintf

### PR DESCRIPTION
**Pull request by JaimeFrey**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by GregThain** [comment]
Note that we are getting Windows build errors:

```
C:\condor\execute\dir_29204\sources\src\condor_daemon_core.V6\daemon_core.cpp(9989,137): error C2664: 'int sprintf(char *const ,const char *const ,...)': cannot convert argument 2 from 'size_t' to 'const char *const ' [C:\condor\execute\dir_29204\userdir\src\condor_utils\condor_extra_objects.vcxproj]
       C:\condor\execute\dir_29204\sources\src\condor_daemon_core.V6\daemon_core.cpp(9989,28): message : Conversion from integral type to pointer type requires reinterpret_cast, C-style cast or parenthesized function-style cast [C:\condor\execute\dir_29204\userdir\src\condor_utils\condor_extra_objects.vcxproj]
       C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt\stdio.h(1764,37): message : see declaration of 'sprintf' 
       
```

**Comment by GregThain** [comment]
There's a bunch of calls to `std::to_string` that aren't qualified with the `std::` prefix.  Can we explicitly call `std::to_string` everywhere? This works today because a lot of HTCondor .cpp files have a `using namespace std;`  but that's bad form.  

When symbols get added to std::, that introduces potential collisions in the future, so I'd like to remove all the "using namespace std" from the code, but let's make that easier in the future by being explicit today.

File: src/classad/collection.cpp [hunk 1]
```diff
@@ -441,16 +436,13 @@ TruncateLog( )
         CondorErrno = ERR_LOG_OPEN_FAILED;
         CondorErrMsg = "when truncating log, failed to open " + tmpLogFileName
             + " errno=";
-        sprintf( buf, "%d", errno );
-        CondorErrMsg += string( buf );
+        CondorErrMsg += std::sto_string( errno );
```
Comment (timtheisen) [inline src/classad/collection.cpp:439]:
```suggestion
        CondorErrMsg += std::sto_string( errno );
```

**Review by timtheisen — DISMISSED** [review]
Oops, a syntax error crept in.

File: src/classad/collection.cpp (hunk 1 above)
Comment (JaimeFrey) [reply src/classad/collection.cpp]: Yeah. Forgot to 'git add' the fix before committing and pushing.

**Review by GregThain — APPROVED** [review]
LGTM.

**Pull request merged**